cmd/kube-proxy/app/server.go

```diff
@@ -368,13 +394,23 @@ func (o *Options) applyDeprecatedHealthzPortToConfig() {
 		o.config.HealthzBindAddress = ""
 		return
 	}
+	host, _, tag, err := GetBindAddressHostPort(o.config.HealthzBindAddress, int(o.healthzPort))
 
-	index := strings.Index(o.config.HealthzBindAddress, ":")
-	if index != -1 {
-		o.config.HealthzBindAddress = o.config.HealthzBindAddress[0:index]
+	if err != nil || tag == "" {
+		klog.Infof("Deprecated Healthz Port To Config failed: %v\n", err)
+		return
+	}
+
+	bindAddress := ""
+	if tag == "ipv4" {
+		bindAddress = fmt.Sprintf("%s:%d", host, o.healthzPort)
 	}
 
-	o.config.HealthzBindAddress = fmt.Sprintf("%s:%d", o.config.HealthzBindAddress, o.healthzPort)
+	if tag == "ipv6" {
+		bindAddress = fmt.Sprintf("[%s]:%d", host, o.healthzPort)
+	}
+
+	o.config.HealthzBindAddress = bindAddress
 }
 
 func (o *Options) applyDeprecatedMetricsPortToConfig() {
```

Comment:
Would be great to dedup `applyDeprecatedMetricsPortToConfig()` and `applyDeprecatedHealthzPortToConfig()` --- they look almost the same :)

Reply:
It can use o.applyDeprecatedBindAddressPortToConfig("healthz", "metrics") to handle both method now.
https://github.com/kubernetes/kubernetes/pull/76320/files#diff-cf93bbed37202b7d58b5841f07ddd89fR412